cmd/frontend/internal/httpapi/graphql.go

```
@@ -83,7 +83,7 @@ func serveGraphQL(schema *graphql.Schema, rlw *graphqlbackend.RateLimitWatcher,
 			traceData.cost = cost
 		}
 
-		if rl, enabled := rlw.Get(); enabled {
+		if rl, enabled := rlw.Get(); enabled && costErr == nil {
```

do we want to surface this error somewhere if it's not nil? eg logs or returning a non-200 status code

---

We log it at debug level above and also add it to honeycomb

---

Hold on, I think there's still a potential panic here if we DO have validation errors since then err will be nil but so will cost..